feature(node_benchmarks): add result comparison

KnifeyMoloko opened the pull request:
This PR adds 2 new features for the recently introduced node benchmarks:
- comparing the current results against previous results for a given node type
- checking if the possible difference between current node benchmarks and the previous runs average is within a defined tolerance margin
- adding node benchmarks to some of our longevity runs

Email table with results example:

![image](https://user-images.githubusercontent.com/13716678/159699289-f431ff91-5da6-4748-a80a-0957e2816449.png)


## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

fgelcer commented on sdcm/report_templates/results_longevity.html:
```diff
@@ -9,6 +9,44 @@ <h3>
+                                <tr>
+                                    <td style="font-weight: bold">{{ key }}</td>
+                                    {% if value is boolean() %}
+                                    {% if value is true() %}
```
nitpick: missing indentation

KnifeyMoloko replied:
Fixed

fgelcer reviewed (commented):
do you have a print screen with the result as an example of it?
also, a single comment about `.html` file indentation, but other than that, LGTM

KnifeyMoloko commented:
> do you have a print screen with the result as an example of it? also, a single comment about `.html` file indentation, but other than that, LGTM

@fgelcer I added an example screenshot in the PR description

fgelcer commented:
> > do you have a print screen with the result as an example of it? also, a single comment about `.html` file indentation, but other than that, LGTM
> 
> @fgelcer I added an example screenshot in the PR description

i see it, and i suggest, to improve readability to limit these average values to have up to 2 decimal

fgelcer commented:
WDYT?

KnifeyMoloko commented:
> > > do you have a print screen with the result as an example of it? also, a single comment about `.html` file indentation, but other than that, LGTM
> > 
> > 
> > @fgelcer I added an example screenshot in the PR description
> 
> i see it, and i suggest, to improve readability to limit these average values to have up to 2 decimal

@fgelcer Good idea. I added the formatting to the `results_longevity.html` template.

fgelcer reviewed (approved):
LGTM.

it will be nice to see the updated result (with 2 decimal), but approved.

enaydanov commented on sdcm/report_templates/results_longevity.html:
```diff
@@ -9,6 +9,44 @@ <h3>
             <li>Instance type: {{ scylla_instance_type }}</li>
             <li>Number of scylladb nodes: {{ number_of_db_nodes }}</li>
 
+            {% if node_benchmarks is not eq("N/A") %}
```
Is it a correct comparison? As I can see `node_benchmarks` can be `None` or a dict.

KnifeyMoloko replied:
I got confused by the `BaseEmailReporter` and it's `build_data_for_report()` method which replaces missing dict values with "N/A". Looks like `{% if node_benchmarks %}` should do it here.

rayakurl reviewed (approved):
Marking as 'approved'  since 
@fgelcer approved it, bit he's currently on a vacation
